src/libNode/Node.cpp

```diff
@@ -441,10 +445,11 @@ bool Node::StartRetrieveHistory(const SyncType syncType,
   GetIpMapping(ipMapping);
 
   if (!ipMapping.empty()) {
-    string pubKey;
-
     for (auto& ds : *m_mediator.m_DSCommittee) {
-      pubKey = DataConversion::SerializableToHexStr(ds.first);
+      string pubKey;
+      if (!DataConversion::SerializableToHexStr(ds.first, pubKey)) {
+        return false;
```

Comment:
Do we want to `continue` here instead?

Reply:
Or maybe add a warning message, as this code seems highly unlikely to be triggered, and if it ever were, then something is seriously wrong.

Reply:
Yeah should be sufficient I guess